### PR TITLE
ET-4553 Modifying GH action

### PR DESCRIPTION
**Pull request by jorge-cr-13**

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -50,43 +59,56 @@ jobs:
...
+      contents: read
     steps:
       - name: Checkout repository
-        uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # v3.1.0
+        uses: actions/checkout@24cb9080177205b6e8c946b17badbe402adc938f # v3.4.0
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@e1e17a757e536f70e52b5a12b2e8d1d1c60e04ef # v2.0.0
+        with:
+          role-to-assume: ${{ secrets.AWS_ECR_REPOSITORY_OIDC_ROLE }}
```
Comment (Robert-Steiner) [inline .github/workflows/deploy.yml:74]: i think we also need to add the repo to the ocid role policy
`organization/eu-central-1/ecr/embedding_service/gh_role/terragrunt.hcl`

Comment (jorge-cr-13) [reply .github/workflows/deploy.yml:74]: Yes, I did this already in the `infra` repo.

Comment (Robert-Steiner) [reply .github/workflows/deploy.yml:74]: have run this workflow?

Comment (jorge-cr-13) [reply .github/workflows/deploy.yml:74]: Yes, It works! :)

**Pull request merged**